src/Core/Joins.cpp

```diff
@@ -87,4 +94,83 @@ JoinKind reverseJoinKind(JoinKind kind)
     return kind;
 }
 
+void serializeJoinKind(JoinKind kind, WriteBuffer & out)
+{
+    uint8_t val = uint8_t(kind);
+    writeIntBinary(val, out);
+}
+
+JoinKind deserializeJoinKind(ReadBuffer & in)
+{
+    uint8_t val;
+    readIntBinary(val, in);
+
+    if (val == uint8_t(JoinKind::Inner))
+        return JoinKind::Inner;
+    if (val == uint8_t(JoinKind::Left))
+        return JoinKind::Left;
+    if (val == uint8_t(JoinKind::Right))
+        return JoinKind::Right;
+    if (val == uint8_t(JoinKind::Full))
+        return JoinKind::Full;
+    if (val == uint8_t(JoinKind::Cross))
+        return JoinKind::Cross;
+    if (val == uint8_t(JoinKind::Comma))
+        return JoinKind::Comma;
+    if (val == uint8_t(JoinKind::Paste))
+        return JoinKind::Paste;
+
+    throw Exception(ErrorCodes::INCORRECT_DATA, "Cannot convert {} to JoinKind", UInt16(val));
+}
+
+void serializeJoinStrictness(JoinStrictness strictness, WriteBuffer & out)
+{
+    uint8_t val = uint8_t(strictness);
+    writeIntBinary(val, out);
+}
+
+JoinStrictness deserializeJoinStrictness(ReadBuffer & in)
+{
+    uint8_t val;
+    readIntBinary(val, in);
+
+    if (val == uint8_t(JoinStrictness::Unspecified))
+        return JoinStrictness::Unspecified;
+    if (val == uint8_t(JoinStrictness::RightAny))
+        return JoinStrictness::RightAny;
+    if (val == uint8_t(JoinStrictness::Any))
+        return JoinStrictness::Any;
+    if (val == uint8_t(JoinStrictness::All))
+        return JoinStrictness::All;
+    if (val == uint8_t(JoinStrictness::Asof))
+        return JoinStrictness::Asof;
+    if (val == uint8_t(JoinStrictness::Semi))
+        return JoinStrictness::Semi;
+    if (val == uint8_t(JoinStrictness::Anti))
+        return JoinStrictness::Anti;
```

Comment:
You can just use `static_cast` here

Reply:
This applies to all the enums here. If you want to be more secure, it's enough to add values to enumerations and check in the deserialize methods that the read value does not exceed the actual maximum

Reply:
Yes, I thought about it as well, but I did not want to add MAX into the enum (this will break switch-case check);